L1Trigger/L1TGlobal/interface/GlobalBoard.h

```diff
@@ -277,6 +271,33 @@ namespace l1t {
 
     // start the PS counter from a random value between [1,PS] instead of PS
     bool m_semiRandomInitialPSCounters = false;
+
+    // precision for fractional prescales
+    static const size_t m_precision = 2;
+
+    // struct to increment the prescale according to fractional prescale logic in firmware
+    struct PrescaleCounter {
+      const size_t prescale_count;
+      const size_t single_step;
+      size_t trigger_counter;
+
+      PrescaleCounter(double prescale, size_t prec, size_t initial_counter = 0)
+          : prescale_count(std::lround(prescale * std::pow(10, prec))),
+            single_step(std::pow(10, prec)),
+            trigger_counter(initial_counter) {}
+
+      // function to increment the prescale counter and return the decision
+      bool accept();
+    };
```

Comment:
Just a minor suggestion: if the `size_t prec` argument is dropped and is assumed to be `m_precision`, `single_step` could also be moved to the parent class and changed to `constexpr`, making `accept()` slightly faster and saving O(8 KB).

Reply:
I would agree. The current generalisation can always be re-introduced later on, if needed.